ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,393 @@
         """
-        This will complete the partial instruction by using the possible codes list. It will attempt at finding the
-        correct form for the instruction.
-        :param partialInstruction: Instruction, the partial instruction that needs to be completed
-        :param possibleCodes: list, a list of possible partials
-        :return: return the instructionCode and the form to be used to further work on the instruction
+        Takes in the form based on instruction and its arguments and determines which form it belongs to.
+        This state contains all possible instructions for that particular form.
+        :param form: str, The form of the instruction based on its arguments (ex: instruction-register-immediate)
+        :return: str, The appropriate state string for the instruction form
         """
 
-        instructionCode = None
-        instructionForm = None
+        try:
+            state = STATE_LIST[form]
 
-        for code in possibleCodes:
-            for form in formDescription:
-                if formDescription[form]["typeCode"] is not (code >> 4):
-                    continue
-                found = True
-                for subElem in formDescription[form]["description"]:
-                    if subElem == "instructionCode":
-                        continue
-                    if getattr(partialInstruction, subElem) is None:
-                        # This is not what we are looking for
-                        found = False
-                        break
-                    else:
-                        found = True
-
-                if found:
-                    instructionCode = code
-                    instructionForm = form
-
-        goodInstructionParams = ["instructionCode",
-                                 "sourceRegister",
-                                 "destinationRegister",
-                                 "sourceImmediate",
-                                 "destinationImmediate",
-                                 "width",
-                                 "flags",
-                                 "instructionLength"]
-
-        # This deals with unnecessarily long instructions. Ex: JMP $A $B <- one too many register
-        for param in goodInstructionParams:
-            if param not in formDescription[instructionForm]["description"]:
-                insParam = getattr(partialInstruction, param)
-                if insParam is not None:
-                    raise ValueError("Invalid instruction provided")
-
-        return instructionCode, instructionForm
-
-    def _generateBinaryInstructionFromInstruction(self, instruction: Instruction=None, form: dict=None):
+        except KeyError as e:
+            # Form is invalid, doesn't fit any state descriptions
+            raise ValueError("Invalid instruction format at line " + str(self.lineno))
+
+        return state
+
+    def _evaluateIndicatorData(self, text, dataIdentifier, line, instruction, labelFlag):
```

Comment:
labelFlag is set in the method but not actually used. I don't think this parameter should be there.